File: atividade-6/atividade.go
package main

import (
	"fmt"
)

type AFN struct {
	states       map[string]bool
	startState   string
	acceptStates map[string]bool
}

func NewAFN() *AFN {
	return &AFN{
		states:       map[string]bool{"q0": true, "q1": true},
		startState:   "q0",
		acceptStates: map[string]bool{"q1": true},
	}
}

func (afn *AFN) isAccepted(input string) bool {
	currentStates := map[string]bool{afn.startState: true}

	for _, char := range input {
		nextStates := make(map[string]bool)
		for state := range currentStates {
			if state == "q0" {
				if char == '0' {
					nextStates["q1"] = true 
				} else {
					nextStates["q0"] = true 
				}
			} else if state == "q1" {
				nextStates["q1"] = true 
			}
		}
		currentStates = nextStates
	}

	for state := range currentStates {
		if afn.acceptStates[state] {
			return true
		}
	}
	return false
}

func main() {
	afn := NewAFN()
	fmt.Println(afn.isAccepted("111"))
	fmt.Println(afn.isAccepted("110"))
	fmt.Println(afn.isAccepted("001"))
	fmt.Println(afn.isAccepted("1"))  
	fmt.Println(afn.isAccepted("0"))  
}

File: atividade-6/atividade.py
class AFN:
    def __init__(self):
        self.states = {'q0', 'q1'}
        self.start_state = 'q0'
        self.accept_states = {'q1'}
    
    def is_accepted(self, string):
        current_states = {self.start_state}  
        
        for char in string:
            next_states = set()
            for state in current_states:
                if state == 'q0':
                    if char == '0':
                        next_states.add('q1')  
                    else:
                        next_states.add('q0')  
                elif state == 'q1':
                    next_states.add('q1')  
            current_states = next_states
        
        return any(state in self.accept_states for state in current_states)

if __name__ == '__main__':
    afn = AFN()
    print(afn.is_accepted("111"))
    print(afn.is_accepted("110"))
    print(afn.is_accepted("001"))
    print(afn.is_accepted("1"))  
    print(afn.is_accepted("0"))  